src/go/k8s/apis/redpanda/v1alpha1/cluster_types.go

```diff
@@ -170,6 +170,13 @@ type ClusterSpec struct {
 
 	// If key is not provided in the SecretRef, Secret data should have key "license"
 	LicenseRef *SecretKeyRef `json:"licenseRef,omitempty"`
+
+	// When InitialValidationForVolume is enabled the mounted Redpanda data folder
+	// will be checked if:
+	// - it is dir
+	// - it has XFS file system
+	// - it has read and write access
+	InitialValidationForVolume *bool `json:"initialValidationForVolume,omitempty"`
```

Comment:
in addition to, should it also actually *write* and read the data instead of validating that the bitflags are set. 

i.e.: `open(), write('hello world" + uuid), flush(), close(), open(), read("hello world" + uuid)`

Reply:
I might wrongly describe the API. My intention was to do what you proposed in https://github.com/redpanda-data/redpanda/pull/9717/files#diff-3a93dbb2eb15a71f67fa8337aa32d81636799a4db9aa0c457bcc8592685182a6R239-R247

I will change the comment.